### PR TITLE
sd-device: validate input parameter after creating sd-device object

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd/sd-device/sd-device.c [hunk 1]
```diff
@@ -242,22 +242,55 @@ _public_ int sd_device_new_from_syspath(sd_device **ret, const char *syspath) {
...
+        r = sd_device_new_from_syspath(&dev, syspath);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_devnum(dev, &n);
+        if (r == -ENOENT)
+                return -ENXIO;
+        if (r < 0)
+                return r;
+        if (major(n) != major(devnum) || minor(n) != minor(devnum))
```
Comment (poettering) [inline src/libsystemd/sd-device/sd-device.c]: hm, `n != devnum` should suffice, no?

**Comment by yuwata** [comment]
@poettering Thank you for the review. The above comment is addressed. Upgrading the green label.

File: src/libsystemd/sd-device/sd-device.c [hunk 2]
```diff
@@ -634,43 +634,29 @@ static int handle_uevent_line(
...
                 *minor = value;
-        else {
-                r = device_add_property_internal(device, key, value);
-                if (r < 0)
-                        return r;
-        }
+        else
+                return device_add_property_internal(device, key, value);
 
         return 0;
```
Comment (keszybz) [inline src/libsystemd/sd-device/sd-device.c:661]: This is now unreachable, please drop.

**Review by keszybz — COMMENTED** [review]
LGTM too.

File: src/libsystemd/sd-device/sd-device.c (hunk 2 above)
Comment (yuwata) [reply src/libsystemd/sd-device/sd-device.c:661]: No, on "MAJOR" or "MINOR" cases, here is reachable.

Comment (keszybz) [reply src/libsystemd/sd-device/sd-device.c:661]: Ah, indeed.

**Pull request merged**